### ISSUE TITLE
CI: Make clippy warnings trigger warnings instead of failures

### ISSUE THREAD

**Issue by Techcable** [slog-rs/slog#338]
For example, while working on PR #337 I triggered the clippy lint  `#[warn(clippy::std-instead-of-alloc)]`

<details>
   <summary>CI error log</summary>

```
error: used import from `std` instead of `alloc`
    --> src/lib.rs:2423:8
     |
2423 |     T: std::string::ToString + FromStr + PartialEq + fmt::Debug,
     |        ^^^^^^^^^^^^^^^^^^^^^
     |
     = help: consider importing the item from `alloc`
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#std_instead_of_alloc
     = note: `-D clippy::std-instead-of-alloc` implied by `-D warnings`

error: used import from `std` instead of `core`
  --> src/tests.rs:37:14
   |
37 |         ) -> std::result::Result<Self::Ok, Self::Err> {
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider importing the item from `core`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#std_instead_of_core
   = note: `-D clippy::std-instead-of-core` implied by `-D warnings`

error: used import from `std` instead of `alloc`
  --> src/tests.rs:69:26
   |
69 |     struct TestError<E = std::string::ParseError>(&'static str, Option<E>);
   |                          ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider importing the item from `alloc`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#std_instead_of_alloc

error: used import from `std` instead of `core`
   --> src/tests.rs:115:18
    |
115 |             ) -> std::result::Result<Self::Ok, Self::Err> {
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: consider importing the item from `core`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#std_instead_of_core
```

</details>

Ideally this would have triggered a github actions [`"warnings::"` annotation](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#setting-a-warning-message) instead of a hard failure.
This would still flag the issue without completely failing the build.

**Comment by Techcable** [comment]
This would be fixed by #339 but seems to have some downsides.

**Comment by Techcable** [comment]
I realized that since we only fail on the "hardcoded recent version" triggering hard errors shouldn't happen unexpectedly due to clippy changes.

The only reason my changes triggered `#[warn(clippy::std-instead-of-alloc)]` was because I explicitly enabled that warning.
